### PR TITLE
docs(translation): translate `useImperativeHandle`

### LINKED ISSUE #403: `useImperativeHandle`
Page URL: https://id.react.dev/reference/react/useImperativeHandle

Assignee: @sozonome

### PR DESCRIPTION
**Pull request by sozonome**
Closes #403

## Description

Translate the `useImperativeHandle` page.
Page URL: https://id.react.dev/reference/react/useImperativeHandle

### CODE REVIEW

File: src/content/reference/react/useImperativeHandle.md
Comment (mazipan) [inline src/content/reference/react/useImperativeHandle.md:41]:
```suggestion
* `ref`: *ref* yang Anda terima sebagai argumen ke-dua dari [fungsi *render* `forwardRef`.](/reference/react/forwardRef#render-function)
```

File: src/content/reference/react/useImperativeHandle.md
Comment (mazipan) [inline src/content/reference/react/useImperativeHandle.md:45]:
```suggestion
* **opsional** `dependencies`: Daftar semua nilai reaktif yang dirujuk di dalam kode `setup`. Nilai reaktif termasuk *props*, *state*, dan semua variabel dan fungsi yang dideklarasikan langsung di dalam komponen. Jika *linter* Anda telah [dikonfigurasi untuk React](/learn/editor-setup#linting), maka *linter* tersebut akan memverifikasi bahwa setiap nilai reaktif sudah diatur dengan benar sebagai dependensi. Daftar dependensi ini harus memiliki jumlah *item* yang konstan dan ditulis secara *inline* seperti `[dep1, dep2, dep3]`. React akan membandingkan setiap dependensi dengan nilai lama menggunakan perbandingan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is). Jika sebuah *render* ulang menghasilkan sebuah perubahan terhadap beberapa *dependency*, atau jika Anda menghilangkan argumen ini, fungsi `createHandle` Anda akan dijalankan ulang, dan penanganan yang baru dibuat akan ditempatkan ke `ref`.
```

File: src/content/reference/react/useImperativeHandle.md
Comment (mazipan) [inline src/content/reference/react/useImperativeHandle.md:47]:
```suggestion
#### Kembalian {/*returns*/}
```

File: src/content/reference/react/useImperativeHandle.md
Comment (mazipan) [inline src/content/reference/react/useImperativeHandle.md:169]: Kita emang menterjemahkan *method* dengan mentoda ya @mhaidarhanif @zainfathoni @r17x @resir014 ?

Comment (sozonome) [reply src/content/reference/react/useImperativeHandle.md:169]: Maaf, stelah cek docs halaman lain, seharusnya tidak perlu di translate, ckup di italic saja *method*

**Review by mazipan — APPROVED** [review]
Looks good to me 🟢, let's wait for review from @mhaidarhanif, @r17x @resir014 or @zainfathoni

File: src/content/reference/react/useImperativeHandle.md
Comment (resir014) [inline src/content/reference/react/useImperativeHandle.md:169]:
```suggestion
### Mengekspos methods imperatif Anda sendiri {/*exposing-your-own-imperative-methods*/}
```

Untuk kata serapan di dalam heading, jangan di-*italic*-kan. ([Referensi](https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Universal#kata-serapan-di-dalam-heading))

**Review by resir014 — DISMISSED** [review]

**Review by r17x — APPROVED** [review]

File: src/content/reference/react/useImperativeHandle.md
Comment (sozonome) [reply src/content/reference/react/useImperativeHandle.md]: Thank you for reminding @resir014 🙇, updated 🙏

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:67]:
```suggestion
Dengan kode di atas, [sebuah *ref* yang diteruskan ke komponen `MyInput` akan menerima simpul DOM node dari `<input>`.](/reference/react/forwardRef#exposing-a-dom-node-to-the-parent-component) Namun, Anda dapat mengekspos sebuah nilai kustom. Untuk mengkustom penanganan yang diekspos, panggil fungsi `useImperativeHandle` di tingkat teratas komponen Anda:
```

1. Translate `DOM node`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#n)
2. Translate `top` to `teratas`, as I proposed at https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5772140.

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:45]:
```suggestion
* `dependencies` **opsional**: Daftar semua nilai reaktif yang dirujuk di dalam kode `setup`. Nilai reaktif termasuk *props*, *state*, dan semua variabel dan fungsi yang dideklarasikan langsung di dalam tubuh komponen. Jika *linter* Anda telah [dikonfigurasi untuk React](/learn/editor-setup#linting), maka *linter* tersebut akan memverifikasi bahwa setiap nilai reaktif sudah diatur dengan benar sebagai dependensi. Daftar dependensi ini harus memiliki jumlah *item* yang konstan dan ditulis secara *inline* seperti `[dep1, dep2, dep3]`. React akan membandingkan setiap dependensi dengan nilai lama menggunakan perbandingan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is). Jika sebuah *render* ulang menghasilkan sebuah perubahan terhadap beberapa *dependency*, atau jika Anda menghilangkan argumen ini, fungsi `createHandle` Anda akan dijalankan ulang, dan penanganan yang baru dibuat akan ditetapkan kepada *ref* tersebut.
```

1. Change "*menerangkan-diterangkan*" pattern in English into "*diterangkan-menerangkan*" pattern in bahasa Indonesia.
2. Retain the `body` word and translate it to `tubuh`.
3. Translate `assigned` to `ditetapkan`, as proposed by @mhaidarhanif at https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5768126.

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:41]:
```suggestion
* `ref`: *ref* yang Anda terima sebagai argumen kedua dari [fungsi *render* `forwardRef`.](/reference/react/forwardRef#render-function)
```

It's either `ke-2` or `kedua`, not `ke-dua`.
[Referensi](https://ejaan.kemdikbud.go.id/eyd/penulisan-kata/angka-dan-bilangan/#9-penulisan-bilangan-tingkat-dapat-menggunakan-angka-romawi-gabungan-awalan-ke-dan-angka-arab-atau-huruf):
image

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:43]:
```suggestion
* `createHandle`: Sebuah fungsi yang tidak mengambil argumen dan mengembalikan penanganan *ref* yang ingin Anda ekspos. Penanganan *ref* tersebut dapat memiliki tipe apapun. Biasanya, Anda akan mengembalikan sebuah objek dengan sekumpulan metode yang ingin Anda ekspos.
```

1. Please italicize `ref`
2. Please translate `object` to `object`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#o)
3. `any` here is not in the context of `any` type literally. `createHandle` is a function, which shouldn't have an `any` type. It could be anything that is a subset of `function`.

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:57]:
```suggestion
Secara bawaan, komponen tidak mengekspos simpul-simpul DOM (*DOM nodes*) kepada komponen induk. Sebagai contoh, jika Anda menginginkan komponen induk dari `MyInput` untuk [memiliki akses](/learn/manipulating-the-dom-with-refs) terhadap simpul DOM dari `<input>`, Anda harus menyertakannya dengan [`forwardRef`:](/reference/react/forwardRef)
```

1. Translate `DOM nodes`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#n)

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:85]:
```suggestion
Sebagai contoh, jika Anda tidak ingin mengekspos keseluruhan simpul DOM dari `<input>`, namun Anda ingin mengekspos dua *methods* yang tersedia di dalamnya: `focus` dan `scrollIntoView`. Untuk melakukan hal tersebut, pertahankan DOM peramban yang asli ke dalam *ref* yang terpisah. Kemudian gunakan `useImperativeHandle` untuk mengekspos sebuah penanganan hanya dengan *methods* yang ingin Anda panggil melalui komponen induk:
```

1. Translate `DOM node`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#n)
2. Retain the plural form of `methods`.
3. Uncodify `DOM`.
4. Retain the `only` word and translate it to `hanya`.

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:108]:
```suggestion
Sekarang, jika komponen induk mendapatkan sebuah *ref* yang merujuk pada `MyInput`, komponen tersebut akan dapat memanggil *method* `focus` dan `scrollIntoView`. Meskipun begitu, komponen induk tersebut tidak akan memiliki akses penuh terhadap simpul DOM `<input>` yang mendasarinya.
```

1. Translate `DOM node`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#n)

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:171]:
```suggestion
*Methods* yang diekspos melalui penanganan imperatif tidak harus sesuai dengan *DOM methods* secara persis. Sebagai contoh, komponen `Post` berikut mengekspos sebuah *method* `scrollAndFocusAddComment` melalui penanganan imperatif. Hal ini memungkinkan induk `Page` menggulir (*scroll*) daftar komentar *dan* memfokuskan bidang input ketika Anda mengklik tombol:
```

1. Retain the original `DOM methods` plural form.
2. Introduce `scroll` translation for the first time. Let's propose it here: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5772140.

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:284]:
```suggestion
**Jangan terlalu sering menggunakan refs.** Anda hanya boleh menggunakan *refs* untuk perilaku *imperatif* yang tidak dapat Anda ungkapkan sebagai *props*: misalnya, menggulir ke sebuah simpul, memfokuskan sebuah simpul, memicu sebuah animasi, memilih teks, dan sebagainya.
```

1. Translate `node`. [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#n)

File: src/content/reference/react/useImperativeHandle.md
Comment (zainfathoni) [inline src/content/reference/react/useImperativeHandle.md:286]:
```suggestion
**Jika Anda dapat mengekspresikan sesuatu sebagai *prop*, Anda tidak seharusnya menggunakan *ref*.** Sebagai contoh, alih-alih mengekspos sebuah penanganan imperatif seperti `{ open, close }` dari sebuah komponen `Modal`, lebih baik menggunakan `isOpen` sebagai *prop* seperti `<Modal isOpen={isOpen} />`. [Efek](/learn/synchronizing-with-effects) dapat membantu Anda mengekspos perilaku imperatif melalui  *props*.
```

1. Translate `should` to `seharusnya`. I'm proposing it here: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5772140
2. Translate `instead of` to `alih-alih`. I'm proposing it here: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5772140
3. Translate `effects` to `efek`. As proposed by @mhaidarhanif here: https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5768126

**Review by zainfathoni — CHANGES_REQUESTED** [review]
I have a few more suggestions while introducing more words to the glossary.

**Comment by sozonome** [comment]
Hi @resir014 @zainfathoni thank you so much for the review and suggestions, updated 🙏

**Comment by r17x** [comment]
Need to re-check @zainfathoni  @resir014

**Comment by mazipan** [comment]
Help to re-review @zainfathoni @resir014

**Comment by sozonome** [comment]
Hi @zainfathoni @resir014 need help to re-review 🙇 🙏

**Review by mazipan — APPROVED** [review]

**Comment by mazipan** [comment]
Merging blocked @resir014 @zainfathoni

**Review by zainfathoni — APPROVED** [review]
LGTM 💯

Thanks for making the changes and sorry for the delayed review. 🙏

**Comment by mazipan** [comment]
@resir014, your call

**Review by mhaidarhanif — APPROVED** [review]
As there are 3 other approvals

**Pull request merged**
